source/example1_sync_fifo.rst

```diff
@@ -1,2 +1,103 @@
+    // 
+    //////////////////////////////////////////////////////////////////////////////////
+    module sync_fifo
+        #(
+            parameter int unsigned DEPTH    = 8             ,
+            parameter int unsigned DWIDTH   = 32            ,
+            parameter int unsigned AWIDTH   = $clog2(DEPTH)   
+        )(
+            input                       clk     ,
+            input                       rst_n   ,
+            // Req
+            input                       req_vld ,
+            output  logic               req_rdy ,
+            input   logic [DWIDTH-1:0]  req_data,
+            // ACK
+            output  logic               ack_vld ,
+            input                       ack_rdy ,
+            output  logic [DWIDTH-1:0]  ack_data
+        );
+
+    //=================================================================
+    // Internal Signal
+    //=================================================================
+        logic [DWIDTH-1:0]  data_mem [DEPTH-1:0];
+        logic [AWIDTH  :0]  wr_ptr              ;
+        logic [AWIDTH  :0]  rd_ptr              ;
+        logic               rd_en               ;
+        logic               wr_en               ;
+        logic               fifo_full           ;
+        logic               fifo_empty          ;
+
+        logic               dummy_wire1;
+        logic               dummy_wire2;
+
+    //=================================================================
+    // WR/RD enable
+    //=================================================================
+        assign wr_en = req_vld && req_rdy;
+        assign rd_en = ack_vld && ack_rdy;
+
+    //=================================================================
+    // Empty & Full
+    //=================================================================
+        assign fifo_empty = wr_ptr == rd_ptr;
+        assign fifo_full  = {~wr_ptr[AWIDTH], wr_ptr[AWIDTH-1:0]} == rd_ptr;
+
+    //=================================================================
+    // Handshake
+    //=================================================================
+        assign ack_vld = ~fifo_empty;
+        assign req_rdy = ~fifo_full;
+
+    //=================================================================
+    // WR/RD control
+    //=================================================================
+        always_ff @(posedge clk or negedge rst_n) begin
+            if (~rst_n) begin
+                wr_ptr <= {AWIDTH+1{1'b0}};
+                rd_ptr <= {AWIDTH+1{1'b0}};
+            end
+            else begin
+                if (rd_en) {dummy_wire1, rd_ptr} <= rd_ptr + 1'b1;
+                if (wr_en) {dummy_wire2, wr_ptr} <= wr_ptr + 1'b1;
+            end
+        end
+
+    //=================================================================
+    // Mem access
+    //=================================================================
+        always_ff @(posedge clk or negedge rst_n) begin
```

Comment:
no need to reset memory, it will make reset recovery/removal timing hard to closure.